pkg/webui/console/store/middleware/logics/events.js

```diff
@@ -131,22 +129,18 @@ const createEventsConnectLogics = (reducerName, entityName, onEventsStart) => {
         const filterRegExp = Boolean(filter) ? filter.filterRegExp : undefined
 
         try {
-          channel = await onEventsStart([id], filterRegExp, EVENT_TAIL, after)
+          const listeners = {
+            message: message => dispatch(getEventSuccess(id, message)),
+            error: error => dispatch(getEventFailure(id, error)),
+            close: wasClientRequest => dispatch(closeEvents(id, { silent: wasClientRequest })),
+          }
+          channel = await onEventsStart([id], filterRegExp, EVENT_TAIL, after, listeners)
           dispatch(startEventsSuccess(id, { silent }))
-
-          channel.on('message', message => dispatch(getEventSuccess(id, message)))
-          channel.on('error', error => dispatch(getEventFailure(id, error)))
-          channel.on('close', wasClientRequest => {
-            dispatch(closeEvents(id, { silent: wasClientRequest }))
-            channel = null
-          })
+          channel.open()
```

Comment:
No events are emitted before `open`, in order to guarantee that things are dispatched in order:
1. Events start success.
2. Events get success x N.
3. Events get failed (maybe).
4. Events closed.